### PR TITLE
fix(timezone): Add timezone awareness to time operations based on UTC.

### PR DESCRIPTION
**Pull request by samiralavi**
PostreSQL datetime type doesn't include the time zone information and the recommended approach is that the application should know the time zone when it stores the data and retrieves it. Aqueduct will now keep the time zone as UTC when stored in the database. Furthermore, the API will be based on UTC time zones. In the API returned values, the serialized datetime comes with time zone information included (i.e. for UTC `+00:00`). The input types such as the date filter input type accepts serialized datetime in any time zone (ISO format); it gets updated to  UTC time zone before being used by the service layer. This PR adds the required tests to consider start and stop datetime.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -52,6 +52,7 @@ alembic = "^1.12"
 psycopg2-binary = "^2.9"
 mkdocstrings = {version = "^0.24",extras = ["python"]}
 nbconvert = "^7.14"
+pytz = "^2024.1"
```
Comment (str-anger) [inline pyproject.toml:55]: I'd disapprove this change. You are bringing a dependency only for the sake of the single test which creates a time zone by the name. I think you might avoid adding a new package, OR, at least make it optional and add to extras, such that the non-dev version won't require adding this library.

Comment (samiralavi) [reply pyproject.toml:55]: I understand your point about dependencies and I agree with that, but this is a development dependency in the `toml` file defined under `[tool.poetry.group.dev.dependencies]`. Could you review this again?

File: aqueductcore/backend/services/experiment.py [hunk 2]
```diff
@@ -90,15 +90,15 @@ async def get_all_experiments(  # pylint: disable=too-many-arguments
         )
 
     if start_date is not None and end_date is not None:
-        start_date = datetime.combine(start_date, time.min)
```
[the remark this answers is not in the document]
Comment (samiralavi) [reply aqueductcore/backend/services/experiment.py:93]: old logic 🗡️ 😆

**Review by str-anger — APPROVED** [review]

File: aqueductcore/backend/services/experiment.py [hunk 3]
```diff
@@ -90,15 +90,15 @@ async def get_all_experiments(  # pylint: disable=too-many-arguments
...
-        end_date = datetime.combine(end_date, time.max)
-        statement = statement.filter(orm.Experiment.created_at.between(start_date, end_date))
+        statement = statement.filter(
+            orm.Experiment.created_at.between(start_date.astimezone(timezone.utc), end_date)
```
Comment (safeamiiir) [inline aqueductcore/backend/services/experiment.py:94]: why are you not also using `end_date.astimezone(timezone.utc)`?

**Review by safeamiiir — APPROVED** [review]

File: aqueductcore/backend/services/experiment.py (hunk 3 above)
Comment (samiralavi) [reply aqueductcore/backend/services/experiment.py:94]: Good catch, thanks, it was missing here! It seems the tests I have written did not cover this bug. Will fix.

**Review by jatinriverlane — APPROVED** [review]

**Pull request merged**